src/gas/fetchGasEstimatesViaEthFeeHistory.ts

```diff
@@ -0,0 +1,164 @@
+import { BN } from 'ethereumjs-util';
+import { fromWei } from 'ethjs-unit';
+import fetchBlockFeeHistory from './fetchBlockFeeHistory';
+import { Eip1559GasFee, GasFeeEstimates } from './GasFeeController';
+
+type EthQuery = any;
+type PriorityLevel = typeof PRIORITY_LEVELS[number];
+type Percentile = typeof PRIORITY_LEVEL_PERCENTILES[number];
+
+// This code is translated from the MetaSwap API:
+// <https://gitlab.com/ConsenSys/codefi/products/metaswap/gas-api/-/blob/017436f628b2d5967f6e8734780a9114f9e58af9/src/eip1559/feeHistory.ts>
```

Comment:
Note that this file is not a direct translation. I rearranged the code because it made more sense to me and was easier to follow when presented this way (also, `fetchBlockFeeHistory` returns data in a different format than `eth_feeHistory`).